sycl/include/sycl/vector.hpp

```diff
@@ -313,13 +222,25 @@ class __SYCL_EBO vec
                         detail::GetOp, Indexes...>;
 
   // Shortcuts for args validation in vec(const argTN &... args) ctor.
-  template <typename... argTN>
-  using EnableIfSuitableTypes = typename std::enable_if_t<
-      std::conjunction_v<TypeChecker<argTN, DataT>...>>;
+  template <typename CtorArgTy>
+  static constexpr bool AllowArgTypeInVariadicCtor = []() constexpr {
+    // FIXME: This logic implements the behavior of the previous implementation.
```

Comment:
Just to clarify, what needs to be fixed here? Is it the following?

> FIXME: Swizzle's `operator[]` for expression trees seems to be broken
>       // and returns values of the underlying vector of some of the operands.

Reply:
They checks for the allowed types, `is_convertible` vs `is_same`. The current spec is `is_same`, we'll probably change it to `is_convertible`, that's why I worded this very opaque so that the comment won't become stale once we do.